index.js

```diff
@@ -1,14 +1,23 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
-		get: function (target, name, receiver) {
+module.exports = function (target) {
+	var handler = {
+		get: function (target, name) {
 			if (!(name in target)) {
 				throw new TypeError('Unknown property: ' + name);
 			}
 
-			return Reflect.get(target, name, receiver);
+			return Reflect.get(target, name);
 		}
-	});
+	};
+
+	var instance = new Proxy(target, handler);
+
+	instance.decorateWith = function (fn) {
```

Comment:
> Can you provide some insight on how to use the @sintax on a context where no transpiling occurs?

It's explained in the spec: 

> A decorator is an expression, that evaluates to a function, that takes the target, name, and decorator descriptor as arguments.

So it's just a simple function that accepts the class and more and does something to it.

Some more examples here: http://babeljs.io/docs/plugins/transform-decorators/

> I'm trying to follow something like this

This is not the same thing.